logic_tree_tools.py

```diff
@@ -0,0 +1,919 @@
+    logic_tree : LogicTree
+        The logic tree for which information is to be printed.
+    """
+
+    print("")  # Add a blank line for readability
+
+    if isinstance(logic_tree, SourceLogicTree):
+        print("Logic tree is a SourceLogicTree")
+
+    if isinstance(logic_tree, GMCMLogicTree):
+        print("Logic tree is a GMCMLogicTree")
+
+    print(f"Logic tree has {len(logic_tree.branch_sets)} branch sets")
+
+    for branch_set_index, branch_set in enumerate(logic_tree.branch_sets):
+
+        print(
+            f"Branch set index {branch_set_index} has name {branch_set.long_name} ({branch_set.short_name}) and contains {len(branch_set.branches)} branches"
+        )
+
+
+def print_info_about_logic_tree_pairs(
+    logic_tree_pairs: Union[list[CustomLogicTreePair], CustomLogicTreePair]
+):
+    """
+    Print information about a logic tree pair or a list of logic tree pairs.
+
+    Parameters
+    ----------
+    logic_tree_pairs : list[CustomLogicTreePair] or CustomLogicTreePair
+        The logic_tree_pairs to print information about.
+    """
+
+    if not isinstance(logic_tree_pairs, list):
+        logic_tree_pairs = [logic_tree_pairs]
+
+    print(f"Printing information about {len(logic_tree_pairs)} logic tree pairs")
+
+    for logic_tree_pair_index, logic_tree_pair in enumerate(logic_tree_pairs):
+        print()
+        print(f"Logic tree at index {logic_tree_pair_index}:")
+        print(
+            f"source_logic_tree_note: {logic_tree_pairs[logic_tree_pair_index].source_logic_tree_note}"
+        )
+        print(
+            f"ground_motion_logic_tree_note: {logic_tree_pairs[logic_tree_pair_index].ground_motion_logic_tree_note}"
+        )
+
+        if logic_tree_pair.source_logic_tree is not None:
+            print_info_about_logic_tree(logic_tree_pair.source_logic_tree)
+
+        if logic_tree_pair.ground_motion_logic_tree is not None:
+            print_info_about_logic_tree(logic_tree_pair.ground_motion_logic_tree)
+
+
+def get_logic_tree_pairs_for_tectonic_selection(
+    initial_logic_tree_pair: CustomLogicTreePair,
+    tectonic_region_type_sets: list[list[str]],
+    which_interfaces: list[str],
+) -> list[CustomLogicTreePair]:
+    """
+    Produces a list of logic tree pairs with the selected tectonic region types.
+
+    Parameters
+    ----------
+    initial_logic_tree_pair : CustomLogicTreePair
+        The initial logic tree pair to select tectonic region types from.
+
+    tectonic_region_type_sets : list[list[str]]
+        A list of lists of the selected tectonic region types for this logic tree pair.
+            Valid tectonic region types are:
+            "Active Shallow Crust",
+            "Subduction Interface",
+            "Subduction Intraslab".
+        Examples:
+         [["Active Shallow Crust"], ["Subduction Interface"], ["Subduction Intraslab"]]
+         [["Active Shallow Crust", "Subduction Interface"], ["Subduction Intraslab"]]
+         ["Subduction Interface"]
```

Comment:
It does not have type `list[list[str]]` it should instead have type `Union[list[str], list[list[str]]]`... Except that is really ugly which is why I suggest that Andrew use `enum.Flag` instead because this is exactly the situation it was designed for.

Reply:
I agree that it would make this typing simpler if an enum.Flag was used because that one object could contain any number of these tectonic region type combinations. However, using enum.Flag would require more modifications than using enum.StrEnum so I used enum.StrEnum everywhere to save time

Reply:
I did correct the type hint in the docstring though. Good spotting.